### PR TITLE
LibWeb: Add support for 'view-box' attribute to SVGs

### LINKED ISSUE #12243: LibWeb: SVG images are often the wrong size
`<svg>` elements can have a `viewBox` attribute as well as a `width` and `height`. This defines a coordinate system that is then used for rendering. (See https://svgwg.org/svg2-draft/coords.html)

It looks like while we do _parse_ the `viewBox`, we don't use it! `SVGSVGElement::view_box()` is never called. I believe what needs to be done is to modify `SVGPathBox::paint()` so that it maps the coordinates (which are relative to the view_box) to screen coordinates (relative to the size of the `<svg>` element).

For example, here is the icon for https://getbootstrap.com in Firefox:
image

And here it is in Browser:
image
Note that it's 118px by 94px, which is exactly the viewBox size:
```
<svg xmlns="http://www.w3.org/2000/svg" width="40" height="32" class="d-block my-1" viewBox="0 0 118 94" role="img">
    <title>Bootstrap</title>
    <path fill-rule="evenodd" clip-rule="evenodd" d="... stuff here ..." fill="currentColor"/></svg>
```

### PR DESCRIPTION
**Pull request by TobyAsE**
One example that doesnt work as in other browsers right now: Having a viewbox with a different aspect ratio than the given width/height doesnt center the svg in the axis that is larger. But that coordinate math is for sometime in the future...

This closes #12243

### CODE REVIEW

File: Userland/Libraries/LibWeb/Layout/SVGGeometryBox.h [hunk 1]
```diff
@@ -21,6 +21,9 @@ class SVGGeometryBox final : public SVGGraphicsBox {
 
     virtual void paint(PaintContext& context, PaintPhase phase) override;
 
+    float viewbox_scaling();
+    Gfx::FloatPoint viewbox_origin();
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/Layout/SVGGeometryBox.h:25]: Can these be `const` qualified?

File: Userland/Libraries/LibGfx/Path.cpp [hunk 2]
```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include "Path.h"
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGfx/Path.cpp:7]: `<LibGfx/Path.h>` is already included below. :^)

File: Userland/Libraries/LibWeb/SVG/SVGSVGElement.h [hunk 3]
```diff
@@ -25,7 +25,7 @@ class SVGSVGElement final : public SVGGraphicsElement {
     virtual bool requires_svg_container() const override { return false; }
     virtual bool is_svg_container() const override { return true; }
 
-    Optional<ViewBox> const& view_box() { return m_view_box; }
+    const Optional<ViewBox>& view_box() const { return m_view_box; }
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/SVG/SVGSVGElement.h:28]: An accidental west-const slipped in here!

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Haven't looked through this thoroughly but I spotted a couple of little things. Very very happy to see this implemented though!

**Pull request merged**